spec.md

```diff
@@ -1080,6 +1081,135 @@ A dictionary value has these methods:
+which elements were first added to the set:
+
+```python
+s = set(["z", "y", "z", "y"])
+len(s)       # prints 2
+s.add("x")
+len(s)       # prints 3
+for e in s:
+    print e  # prints "z", "y", "x"
+```
+
+A set used in Boolean context is true if and only if it contains one or more
+elements.
+
+```python
+s = set()
+"non-empty" if s else "empty"  # "empty"
+t = set(["x", "y"])
+"non-empty" if t else "empty"  # "non-empty"
+```
+
+Sets may be compared for equality using `==` and `!=`. A set `s` is equal to `t`
+if and only if `t` is a set containing elements equal to the elements of `s`,
+but possibly in a different iteration order. In particular, a set is *not* equal
+to the list of its elements. Sets are not ordered with respect to other sets,
+and an attempt to compare two sets using `<`, `<=`, `>`, `>=`, or to sort a
+sequence of sets, will fail.
+
+```python
+set() == set()              # True
+set() != []                 # True
+set([1, 2]) == set([2, 1])  # True
+set([1, 2]) != [1, 2]       # True
+```
+
+The `|` operation on two sets returns the union of the two sets: a set
+containing the elements found in either one or both of the original sets. The
+`|` operation has an augmented assignment version; `s |= t` adds to `s` all the
+elements of `t`.
+
+```python
+set([1, 2]) | set([3, 2])  # set([1, 2, 3])
+s = set([1, 2])
+s |= set([2, 3, 4])        # s now equals set([1, 2, 3, 4])
+```
+
+The `&` operation on two sets returns the intersection of the two sets: a set
+containing only the elements found in both of the original sets. The `&`
+operation has an augmented assignment version; `s &= t` removes from `s` all the
+elements not found in `t`.
+
+```python
+set([1, 2]) & set([2, 3])  # set([2])
+set([1, 2]) & set([3, 4])  # set()
+s = set([1, 2])
+s |= set([0, 1])           # s now equals set([1])
+```
+
+The `-` operation on two sets returns the difference of the two sets: a set
+containing the elements found in the left-hand side set but not the right-hand
+site set. The `-` operation has an augmented assignment version; `s -= t`
+removes from `s` all the elements found in `t`.
+
+```python
+set([1, 2]) - set([2, 3])  # set([1])
+set([1, 2]) & set([3, 4])  # set([1, 2])
+s = set([1, 2])
+s |= set([0, 1])           # s now equals set([2])
+```
+
+The `^` operation on two sets returns the symmetric difference of the two sets:
+a set containing the elements found in exactly one of the two original sets, but
+not in both. The `^` operation has an augmented assignment version; `s ^= t`
+removes from `s` any element of `t` found in `s` and adds to `s` any element of
+`t` not found in `s`.
+
+```python
+set([1, 2]) ^ set([2, 3])  # set([1, 3])
+set([1, 2]) & set([3, 4])  # set([1, 2, 3, 4])
+s = set([1, 2])
+s ^= set([0, 1])           # s now equals set([2, 0])
+```
+
+Like all mutable values in Starlark, a set can be frozen, and once frozen, all
+subsequent operations that attempt to update it will fail.
+
```

Comment:
We should add :

> Sets are instantiated by calling the built-in set function, which returns a set containing all the elements of its optional argument, which must be an iterable sequence. Sets have no literal syntax.

Reply:
Added to the third paragraph of the section.